packages/chain/node.go

```diff
@@ -843,17 +842,25 @@ func (cni *chainNodeImpl) deriveActiveAccessNodes() {
 func (cni *chainNodeImpl) updateAccessNodes(update func()) {
 	cni.accessLock.Lock()
 	oldAccessNodes := cni.activeAccessNodes
+	oldCommitteeNodes := cni.activeCommitteeNodes
 	update()
 	cni.deriveActiveAccessNodes()
 	serverNodes := cni.serverNodes
 	activeAccessNodes := cni.activeAccessNodes
 	activeCommitteeNodes := cni.activeCommitteeNodes
 	cni.accessLock.Unlock()
-	if oldAccessNodes == nil || !util.Same(oldAccessNodes, activeAccessNodes) {
+	anSame := util.Same(oldAccessNodes, activeAccessNodes)
+	cnSame := util.Same(oldCommitteeNodes, activeCommitteeNodes)
+	if !anSame {
 		cni.log.Infof("Access nodes updated, active=%+v", activeAccessNodes)
+		cni.listener.AccessNodesUpdated(cni.chainID, activeAccessNodes)
+	}
+	if !anSame || !cnSame {
+		if !cnSame {
+			cni.log.Infof("Committee nodes updated, active=%+v", activeCommitteeNodes)
+		}
 		cni.mempool.AccessNodesUpdated(activeCommitteeNodes, activeAccessNodes)
 		cni.stateMgr.ChainNodesUpdated(serverNodes, activeAccessNodes, activeCommitteeNodes)
```

Comment:
```suggestion
	anSame := util.Same(oldAccessNodes, activeAccessNodes)
	cnSame := util.Same(oldCommitteeNodes, activeCommitteeNodes)
	if !anSame {
		cni.log.Infof("Access nodes updated, active=%+v", activeAccessNodes)
		cni.listener.AccessNodesUpdated(cni.chainID, activeAccessNodes)
		cni.mempool.AccessNodesUpdated(activeCommitteeNodes, activeAccessNodes)
	}
	if !cnSame {
		cni.log.Infof("Committee nodes updated, active=%+v", activeCommitteeNodes)
		cni.stateMgr.ChainNodesUpdated(serverNodes, activeAccessNodes, activeCommitteeNodes)
```
couldn't we do it like this?

Reply:
Because, for example, `cni.stateMgr.ChainNodesUpdated` has to be called if access nodes OR committee nodes are changed. I decided it is better to duplicate checks than the actual calls to make fewer mistakes.